test-kit/src/test/scala/CrazyhouseDataTest.scala

```diff
@@ -45,7 +45,7 @@ class CrazyhouseDataTest extends ScalaCheckSuite:
       }
       val result = ps
         .filter(_.role != King)
-        .foldLeft(Option(data)) { (data, piece) =>
```

Comment:
yeah, we should keep `Option` here, as compiler doens't know how to infer `Some(data)` as `Option[Data]` 🤦